src/classes/RD_RecurringDonations.cls

```diff
@@ -1214,4 +1538,94 @@ public class RD_RecurringDonations {
+                || newRD.npe03__Contact__c != oldRD.npe03__Contact__c
+                || newRD.npe03__Amount__c != oldRD.npe03__Amount__c
+                || newRD.npe03__Installment_Period__c != oldRD.npe03__Installment_Period__c
+                || newRD.npe03__Installments__c != oldRD.npe03__Installments__c
+                || newRD.Always_Use_Last_Day_Of_Month__c != oldRD.Always_Use_Last_Day_Of_Month__c
+                || newRD.npe03__Schedule_Type__c != oldRD.npe03__Schedule_Type__c
+                || newRD.npe03__Open_Ended_Status__c != oldRD.npe03__Open_Ended_Status__c
+                || newRD.npe03__Next_Payment_Date__c != oldRD.npe03__Next_Payment_Date__c
+                || newRD.npe03__Recurring_Donation_Campaign__c != oldRD.npe03__Recurring_Donation_Campaign__c
+                ) {
+            return true;
+        }
+
+        for (String s : customFieldMappingSettings.keySet()) {
+            npe03__Custom_Field_Mapping__c cfm = customFieldMappingSettings.get(s);
+            if(newRD.get(cfm.npe03__Recurring_Donation_Field__c) != oldRD.get(cfm.npe03__Recurring_Donation_Field__c)) {
+                return true;
+            }
+        }
+
+        return false;
+    }
+
+    /*******************************************************************************************************
+    * @description Creates a new Opportunity on a given date for a given RD
+    * @param oppCloseDate The close date for the new Opportunity
+    * @param r The parent Recurring Donation
+    * @param rdcounter Opportunity counter for naming and Recurring_Donation_Installment_Number__c
+    * @return Opportunity
+    */
+    private static Opportunity newOpportunityOnDate(Date oppCloseDate, npe03__Recurring_Donation__c r, Integer rdcounter) {
+
+        npe03__Recurring_Donations_Settings__c rds = UTIL_CustomSettingsFacade.getRecurringDonationsSettings();
+
+        Opportunity o = new Opportunity();
+        o.CloseDate = oppCloseDate;
+        o.OwnerId = r.OwnerId;
+
+        if(isMultiCurrency){
+            //set the donation currency field to equal the recurring donation currency
+            o.put(OppCurrencyField,r.get(RDCurrencyField));
+        }
+
+        o.Amount = r.npe03__Amount__c;
+
+        String oName = '';
+        oName += r.npe03__Donor_Name__c;
+        oName += ' ';
+        oName += system.label.npe03.RecurringDonationPrefix;
+        oName += ' (';
+        oName += String.valueOf(rdcounter);
+        if(r.npe03__Open_Ended_Status__c == null) {
+            oName += ' of ';
+            oName += String.valueOf(r.npe03__Installments__c);
+        }
+        oName += ') ';
+        oName += o.CloseDate.format();
+        o.Name = oName;
+
+        o.Recurring_Donation_Installment_Number__c = rdcounter;
+
+        if (rds.npe03__Record_Type__c != null){
+            o.put('RecordTypeId', rds.npe03__Record_Type__c);
+            //o.RecordTypeID = rds.npe03__Record_Type__c;
```

Comment:
Why not use `o.RecordTypeID = rds.npe03__Record_Type__c` for this?

Reply:
I'm not sure we can assume recordTypeId exists on opp.  I think if no recordtypes have been defined, then the field doesn't exist, thus why this code uses put().

Reply:
I'm almost positive that most of the codebase assumes that there are Opportunity RecordTypes defined; or at least that the RecordTypeId field exists. I did a search for "RecordTypeId" and there are lots and lots of results where that field is accessed on the Opportunity object.

Reply:
if you are seeing other npsp code using opp.RecordTypeId, then I stand corrected!  thanks for checking.  I only looked in the RD file.

Reply:
Fixed to use o.RecordTypeID = rds.npe03__Record_Type__c; since we can now assume as have RTs configured.